frontend/src/pages/article/components/article-form/article-form.tsx

```diff
@@ -76,38 +76,40 @@ const ArticleForm: React.FC<Properties> = ({ articleForUpdate }) => {
   );
 
   const handleArticleUpdate = useCallback(
-    (payload: ArticleRequestDto): void => {
-      if (!articleForUpdate) {
-        return;
-      }
-
-      const updatePayload = {
-        articleId: articleForUpdate.id,
-        articleForUpdate: {
-          text: payload.text,
-          title: payload.title,
-          publishedAt: articleForUpdate.publishedAt ?? new Date().toISOString(),
-          coverId: payload.coverId,
-        },
-      };
-
-      void dispatch(articlesActions.updateArticle(updatePayload));
-    },
+    (articleSubmitType: ValueOf<typeof ArticleSubmitType>) =>
+      (payload: ArticleRequestDto): void => {
+        if (!articleForUpdate) {
+          return;
+        }
+        const isArticleDrafted = articleSubmitType === ArticleSubmitType.DRAFT;
+
+        const updatePayload = {
+          articleId: articleForUpdate.id,
+          articleForUpdate: {
+            text: payload.text,
+            title: payload.title,
+            publishedAt: isArticleDrafted
+              ? null
+              : articleForUpdate.publishedAt ?? new Date().toISOString(),
+            coverId: payload.coverId,
+          },
+        };
+
+        void dispatch(articlesActions.updateArticle(updatePayload));
+      },
 
     [dispatch, articleForUpdate],
   );
 
   const handleFormSubmit = useCallback(
     (event_: React.BaseSyntheticEvent<SubmitEvent>): void => {
       const button = event_.nativeEvent.submitter as HTMLButtonElement;
+      const submitType = button.name as ValueOf<typeof ArticleSubmitType>;
+      const handleFunction = articleForUpdate
+        ? handleArticleUpdate
+        : handleArticleSubmit;
```

Comment:
Let's split handlers explicitly
```suggestion
      if (articleForUpdate) {
        void handleSubmit(handleArticleUpdate(submitType))(event_);
        return;
      }
      
      void handleSubmit(handleArticleSubmit(submitType))(event_);
```